src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
```

Comment:
PullProofAI

The logic for constructing the response object seems incorrect. The conditions for setting `error`, `message`, and `data` fields are inconsistent and may lead to unexpected behavior. It's better to revise the logic to ensure correct assignment of these fields based on the actual data structure.

```diff
-           error: data && data.message ? data.message : '',
-           message: data && data.message ? data.message : '',
-           data: data && data.message ? { ...data, message: undefined } : data,
+           error: data && data.error ? data.error : '',
+           message: data && data.message ? data.message : '',
+           data: data && data.message ? { ...data, message: undefined, error: undefined } : data,
```

#### Performance Improvement

It's recommended to avoid unnecessary ternary operations and improve readability by simplifying the logic for constructing the response object. Consider refactoring the code as follows:

```typescript
map((data) => {
    const isSuccess = statusCode >= 200 && statusCode < 300;
    const { message, error, ...restData } = data || {};

    const response = {
        isSuccess,
        statusCode,
        error: error || '',
        message: message || '',
        data: restData,
    };

    this.logger.debug(`Response: [${JSON.stringify(response)}] || User Details: [${JSON.stringify(userDetail)}]`);

    return response;
}),
```

This refactored code enhances clarity and maintains consistency in handling response data.


<!-- This is an auto-generated comment by PullProofAI -->